__tests__/cel.test.ts

```diff
@@ -152,15 +152,274 @@ describe("Performance measurements", () => {
+    expect(result).toEqual([true, true, true, 3, 27, 12, 1]);
+  });
+});
+
+describe("Travel Reservation Rules", () => {
+  describe("Premium Discount Eligibility", () => {
+    // Tests a complex pricing rule that considers:
+    // - Total package cost across flight, hotel, and car
+    // - Customer loyalty tier
+    // - Seasonal booking (summer months)
+    const expr = `
+        // Calculate total package cost
+        double(reservation.flight.price) +
+        double(reservation.hotel.nightlyRate) * int(reservation.hotel.nights) +
+        double(reservation.car.dailyRate) * int(reservation.car.days) >= 2000.0 &&
+        // Check loyalty tier
+        reservation.customer.loyaltyTier in ['GOLD', 'PLATINUM'] &&
+        // Check if booking is in summer months (0-based: 5=June, 6=July, 7=August)
+        timestamp(reservation.bookingDate).getMonth() in [5, 6, 7]
+      `;
+    let program: CelProgram;
+
+    beforeEach(async () => {
+      program = await CelProgram.compile(expr);
+    });
+
+    it("should qualify for premium discount with valid summer booking", async () => {
+      const result = await program.execute({
+        reservation: {
+          flight: { price: 1000.0 },
+          hotel: { nightlyRate: 200.0, nights: 4 },
+          car: { dailyRate: 100.0, days: 4 },
+          customer: { loyaltyTier: "PLATINUM" },
+          bookingDate: "2025-07-15T00:00:00Z",
+        },
+      });
+      expect(result).toBe(true);
+    });
+
+    it("should not qualify outside summer months", async () => {
+      const result = await program.execute({
+        reservation: {
+          flight: { price: 1000.0 },
+          hotel: { nightlyRate: 200.0, nights: 4 },
+          car: { dailyRate: 100.0, days: 4 },
+          customer: { loyaltyTier: "PLATINUM" },
+          bookingDate: "2025-12-15T00:00:00Z",
+        },
+      });
+      expect(result).toBe(false);
+      program = await CelProgram.compile(expr);
+    });
+  });
+
+  describe("Multi-condition Booking Validation", () => {
+    // Tests complex booking validation rules that ensure:
+    // - All required components are present
+    // - Logical time sequence of events
+    // - Location consistency
+    // - Capacity constraints
+    const expr = `
+      has(reservation.flight) &&
+      timestamp(reservation.flight.departureTime) < timestamp(reservation.hotel.checkIn) &&
+      timestamp(reservation.hotel.checkIn) < timestamp(reservation.hotel.checkOut) &&
+      (timestamp(reservation.hotel.checkOut) - timestamp(reservation.hotel.checkIn)) > duration("1h") &&
+      timestamp(reservation.hotel.checkOut) < timestamp(reservation.flight.returnTime) &&
+      (reservation.car.pickupLocation == reservation.flight.arrivalAirport ||
+       reservation.car.pickupLocation == reservation.hotel.address.city) &&
+      size(reservation.travelers) <= reservation.hotel.maxOccupancy &&
+      size(reservation.travelers) <= reservation.car.capacity
```

Comment:
cel-rust appears to not support the array.min() function that would simplify this

```
size(res.travelers) <= [a, b].min()
```